### ISSUE TITLE
ratelimit by token scope/claim

### ISSUE THREAD

**Issue by szuecs** [zalando/skipper#629]
Ratelimit supports right now the lookup of X-Forwarded-For header (or client IP) or Authorization header.
Authorization header is a simple way to limit by Bearer Token.
The problem with the Authorization header is that you can circumvent this by getting a fresh token.

A solution would be to lookup a scope from the token and an identifier of the source application.
These have to be set within a filter, to allow applications to ratelimit by source application and enable them to serve different categories, think bronze, silver, gold, to "sell" better API quality.

**Comment by AlexanderYastrebov** [comment]
Probably could be hacked around with `JWTPayloadAnyKVRegexp` predicate and specific rate-limited routes

**Comment by szuecs** [comment]
Yes for the quality idea, but not for refreshing the token which changes auth header.

**Comment by AlexanderYastrebov** [comment]
I mean that by having specific route with `JWTPayloadAnyKVRegexp` users can isolate particular client(+scopes) and then impose `clusterRatelimit` on it without relying on `Authorization` header (though it must be validated).
